### PR TITLE
Banned user can not enter the chat unreliability fix

### PR DESCRIPTION
**Pull request by sczerwinski-wikia**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 1]
```diff
@@ -279,7 +282,7 @@ public void bannedUserCanNotEnterTheChat() {
     }
     finally {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: MAJOR Move this "finally" on the same line that the previous closing curly brace. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ChatTests.java#L281](https://github.com/Wikia/selenium-tests/blob/7210e2079074c69f7ef496c9abf63e890f2aeb1e/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L281): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)
1. ![CRITICAL][CRITICAL] [ChatTests.java#L291](https://github.com/Wikia/selenium-tests/blob/7210e2079074c69f7ef496c9abf63e890f2aeb1e/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L291): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ChatTests.java#L281](https://github.com/Wikia/selenium-tests/blob/b6c0b3e717c016db333c799ab4773cd9f06f846e/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L281): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)
1. ![CRITICAL][CRITICAL] [ChatTests.java#L291](https://github.com/Wikia/selenium-tests/blob/b6c0b3e717c016db333c799ab4773cd9f06f846e/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L291): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 2]
```diff
@@ -276,10 +279,9 @@ public void bannedUserCanNotEnterTheChat() {
       Assertion.assertTrue(chatUserToBeBanned.isPermissionsErrorTitleDisplayed(), "PERMISSION ERROR IS NOT DISPLAYED");
     } catch (InterruptedException e) {
       e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 2 critical

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ChatTests.java#L290](https://github.com/Wikia/selenium-tests/blob/e9df24513a9f5ae15069400ca68dd55a224cb7c4/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L290): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: src/test/java/com/wikia/webdriver/common/core/helpers/User.java [hunk 3]
```diff
@@ -40,7 +40,8 @@
   FORGOTTEN_PASSWORD_SPACES("ci.user.forgottenPasswordSpaces.username", "ci.user.forgottenPasswordSpaces.password"),
   MW119_ADMINISTRATOR("ci.user.mw_admin.username", "ci.user.mw_admin_password", "ci.user.mw_admin.user_id", "ci.user.mw_admin.access_token"),
   CURATED_CONTENT_USER("ci.user.curated_content.username", "ci.user.curated_content.password", "ci.user.curated_content.id", "ci.user.curated_content.access_token"),
-  INFOBOX_BUILDER_ADMIN("ci.user.infobox_builder.username", "ci.user.infobox_builder.password", "ci.user.infobox_builder.id", "ci.user.infobox_builder.access_token");
+  INFOBOX_BUILDER_ADMIN("ci.user.infobox_builder.username", "ci.user.infobox_builder.password", "ci.user.infobox_builder.id", "ci.user.infobox_builder.access_token"),
+  CHAT_USER_TO_BE_BANNED("ci.user.chat.user_to_be_banned.username", "ci.user.chat.user_to_be_banned.password", "ci.user.chat.user_to_be_banned.user_id", "ci.user.chat.user_to_be_banned.access_token");
```
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/common/core/helpers/User.java]: In `ci.user.chat.user_to_be_banned.user_id` there should be `user_id` or `id` ? I honestly don't know

Comment (sczerwinski-wikia) [inline src/test/java/com/wikia/webdriver/common/core/helpers/User.java]: The part from config
<chat>
				<user_to_be_banned>
					<username>ChatUserToBeBanned</username>
					<user_id>32283286</user_id>
					<password>R00mbaP00mba123</password>
					<access_token>P4AvPZC2TYW7zdaz7FO8-Q</access_token>
				</user_to_be_banned>
			</chat>

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 2 critical

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ChatTests.java#L290](https://github.com/Wikia/selenium-tests/blob/fe99c6c37d7ffcc4e7c8496359da4a197772280c/src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java#L290): Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: src/test/java/com/wikia/webdriver/common/core/helpers/User.java (hunk 3 above)
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/common/core/helpers/User.java]: Cool, thanks!

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 4]
```diff
@@ -49,7 +52,11 @@ private ChatPage openChatForUser(String userName, String password) {
     return new ChatPage().open();
   }
 
-  @Test
+  private void unblockUser(String user){
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: MAJOR Remove this unused private "unblockUser" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 5]
```diff
@@ -254,42 +261,26 @@ public void userCanBeKickedOutFromChat() {
...
   }
 
-  @Test
+  @Test(groups = {"ChatTestsForUser_012"})
   public void bannedUserCanNotEnterTheChat() {
+
     ChatPage chatUserStaff = openChatForUser(userStaff, userStaffPassword);
 
     switchToWindow(1);
     SpecialVersionPage chatWindow = new SpecialVersionPage().open();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: MAJOR Remove this unused "chatWindow" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 6]
```diff
@@ -50,7 +52,11 @@ private ChatPage openChatForUser(String userName, String password) {
     return new ChatPage().open();
   }
 
-  @Test
+  private void unblockUser(String user){
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: MAJOR Remove this unused private "unblockUser" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)

File: src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java [hunk 7]
```diff
@@ -255,41 +261,25 @@ public void userCanBeKickedOutFromChat() {
...
   }
 
-  @Test
+  @Test(groups = {"ChatTestsForUser_012"})
   public void bannedUserCanNotEnterTheChat() {
+
     ChatPage chatUserStaff = openChatForUser(userStaff, userStaffPassword);
 
     switchToWindow(1);
     SpecialVersionPage chatWindow = new SpecialVersionPage().open();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/chattests/ChatTests.java]: MAJOR Remove this unused "chatWindow" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**